src/main/scala/org/broadinstitute/dsde/workbench/sam/dataAccess/PostgresDirectoryDAO.scala

```diff
@@ -1355,11 +1343,57 @@ class PostgresDirectoryDAO(protected val writeDbRef: DbReference, protected val
       val userAttributesTable = UserAttributesTable.syntax
       val userAttributesColumns = UserAttributesTable.column
       samsql"""
-        insert into ${UserAttributesTable as userAttributesTable} (${userAttributesColumns.samUserId}, ${userAttributesColumns.marketingConsent}, ${userAttributesColumns.updatedAt})
-          values (${userAttributes.userId}, ${userAttributes.marketingConsent}, ${Instant.now()})
+        insert into ${UserAttributesTable as userAttributesTable} (
+        ${userAttributesColumns.samUserId},
+        ${userAttributesColumns.marketingConsent},
+        ${userAttributesColumns.firstName},
+        ${userAttributesColumns.lastName},
+        ${userAttributesColumns.organization},
+        ${userAttributesColumns.contactEmail},
+        ${userAttributesColumns.title},
+        ${userAttributesColumns.department},
+        ${userAttributesColumns.interestInTerra},
+        ${userAttributesColumns.programLocationCity},
+        ${userAttributesColumns.programLocationState},
+        ${userAttributesColumns.programLocationCountry},
+        ${userAttributesColumns.researchArea},
+        ${userAttributesColumns.additionalAttributes},
+        ${userAttributesColumns.createdAt},
+        ${userAttributesColumns.updatedAt}
+        ) values (
+        ${userAttributes.userId},
+        ${userAttributes.marketingConsent},
+        ${userAttributes.firstName},
+        ${userAttributes.lastName},
+        ${userAttributes.organization},
+        ${userAttributes.contactEmail},
+        ${userAttributes.title},
+        ${userAttributes.department},
+        ARRAY[${userAttributes.interestInTerra.getOrElse(Array.empty[String])}]::text[],
```

Comment:
I had to do a lot of experimenting to get the types correct, this finally didn't throw any type errors